osu.Game/Screens/Play/HUD/SpectatorList.cs

```diff
@@ -94,7 +95,9 @@ protected override void LoadComplete()
             ((IBindableList<SpectatorUser>)watchingUsers).BindTo(client.WatchingUsers);
             ((IBindable<LocalUserPlayingState>)userPlayingState).BindTo(gameplayState.PlayingState);
 
-            watchingUsers.BindCollectionChanged(onSpectatorsChanged, true);
+            watchingUsers.BindCollectionChanged(onWatchingUsersChanged, true);
+            multiplayerClient.RoomUpdated += removePlayersFromMultiplayerRoom;
```

Comment:
I'm a little bit worried about using `RoomUpdated`. I realise that this isn't something I've explicitly brought up before, but it's invoked when _any_ event in the room occurs, including user state changes, beatmap availability changes, user mods and style changes.

In this case we're doing potentially a somewhat substantial amount of allocations during gameplay, and so I think it may have some importance to address?